src/EditorFeatures/CSharpTest/CodeLens/CSharpCodeLensTests.cs

```diff
@@ -0,0 +1,189 @@
+    }
+
+    void D()
+    {
+        C();
+    }
+}"))
+            {
+                var solution = workspace.CurrentSolution;
+                var documentId = workspace.Documents.First().Id;
+                var document = solution.GetDocument(documentId);
+                var syntaxNode = await document.GetSyntaxRootAsync();
+                var iterator = syntaxNode.ChildNodes().First().ChildNodes().GetEnumerator();
+                iterator.MoveNext();
+
+                var result =
+                    await
+                        _referenceService.GetReferenceCountAsync(solution, documentId, iterator.Current,
+                            CancellationToken.None);
+                Assert.True(result.HasValue);
+                Assert.Equal(0, result.Value.Count);
+                Assert.False(result.Value.IsCapped);
+
+                iterator.MoveNext();
+                result =
+                    await
+                        _referenceService.GetReferenceCountAsync(solution, documentId, iterator.Current,
+                            CancellationToken.None);
+                Assert.True(result.HasValue);
+                Assert.Equal(2, result.Value.Count);
+                Assert.False(result.Value.IsCapped);
+
+                iterator.MoveNext();
+                result =
+                    await
+                        _referenceService.GetReferenceCountAsync(solution, documentId, iterator.Current,
+                            CancellationToken.None);
+                Assert.True(result.HasValue);
+                Assert.Equal(1, result.Value.Count);
+                Assert.False(result.Value.IsCapped);
+            }
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeLens)]
+        public async Task TestCapping()
+        {
+            using (var workspace = await SetupWorkspaceAsync(@"
+class A
+{
+    void B()
+    {
+        C();
+    }
+
+    void C()
+    {
+        D();
+    }
+
+    void D()
+    {
+        C();
+    }
+}"))
+            {
+                var solution = workspace.CurrentSolution;
+                var documentId = workspace.Documents.First().Id;
+                var document = solution.GetDocument(documentId);
+                var syntaxNode = await document.GetSyntaxRootAsync();
+                var iterator = syntaxNode.ChildNodes().First().ChildNodes().GetEnumerator();
+                iterator.MoveNext();
+
+                var result =
+                    await
+                        _referenceService.GetReferenceCountAsync(solution, documentId, iterator.Current,
+                            CancellationToken.None, 1);
+                Assert.True(result.HasValue);
+                Assert.Equal(0, result.Value.Count);
+                Assert.False(result.Value.IsCapped);
+
+                iterator.MoveNext();
+                result =
+                    await
+                        _referenceService.GetReferenceCountAsync(solution, documentId, iterator.Current,
```

Comment:
named parameter for 1? and test seems quite repeated? how about creating helper method?